creator-node/src/components/tracks/TranscodeDelegationManager.js

```diff
@@ -0,0 +1,436 @@
+    res = await TranscodeDelegationManager.fetchTranscode(
+      sp,
+      transcodeFileName,
+      fileNameNoExtension
+    )
+    await Utils.writeStreamToFileSystem(res.data, transcodeFilePath)
+
+    // Get m3u8 file and write to tmp disk
+    logger.info({ sp, m3u8FilePath }, 'Fetching m3u8...')
+    const m3u8FileName = fileNameNoExtension + '.m3u8'
+    res = await TranscodeDelegationManager.fetchM3U8File(
+      sp,
+      m3u8FileName,
+      fileNameNoExtension
+    )
+    await Utils.writeStreamToFileSystem(res.data, m3u8FilePath)
+
+    return {
+      transcodeFilePath,
+      segmentFileNames,
+      m3u8FilePath
+    }
+  }
+
+  static async sendTranscodeAndSegmentRequest({
+    sp,
+    fileDir,
+    fileName,
+    fileNameNoExtension
+  }) {
+    const originalTrackFormData =
+      await TranscodeDelegationManager.createFormData(fileDir + '/' + fileName)
+
+    // TODO: make this constant. perhaps in a class
+    const spID = config.get('spID')
+    const { timestamp, signature } =
+      generateTimestampAndSignatureForSPVerification(spID, DELEGATE_PRIVATE_KEY)
+
+    const resp = await this.asyncRetry({
+      asyncFn: axios,
+      asyncFnParams: {
+        url: `${sp}/transcode_and_segment`,
+        method: 'post',
+        data: originalTrackFormData,
+        headers: {
+          ...originalTrackFormData.getHeaders()
+        },
+        params: {
+          uuid: fileNameNoExtension,
+          timestamp,
+          signature,
+          spID
+        },
+        adapter: require('axios/lib/adapters/http'),
+        // Set content length headers (only applicable in server/node environments).
+        // See: https://github.com/axios/axios/issues/1362
+        maxContentLength: Infinity,
+        maxBodyLength: Infinity
+      },
+      asyncFnTask: 'transcode and segment'
+    })
+
+    return resp.data.data.uuid
+  }
+
+  static async createFormData(pathToFile) {
+    const fileExists = await fsExtra.pathExists(pathToFile)
+    if (!fileExists) {
+      throw new Error(`File does not exist at path=${pathToFile}`)
+    }
+
+    const formData = new FormData()
+    formData.append('file', fs.createReadStream(pathToFile))
+
+    return formData
+  }
+
+  static async fetchHealthCheck(sp) {
+    await axios({
+      url: `${sp}/health_check`,
+      method: 'get'
+    })
+  }
+
+  /**
+   * Gets the status of the transcode processing
+   * @param {Object} param
+   * @param {string} param.sp the current sp selected for track processing
+   * @param {string} param.uuid the uuid of the track transcoding task
+   * @returns the status, and the success or failed response if the task is complete
+   */
+  static async fetchTranscodeProcessingStatus({ sp, uuid }) {
+    const spID = config.get('spID')
+    const { timestamp, signature } =
+      generateTimestampAndSignatureForSPVerification(spID, DELEGATE_PRIVATE_KEY)
+
+    const { data: body } = await this.asyncRetry({
```

Comment:
`asyncRetry` feels like more of a `Util` to me wdyt?

Reply:
hmmmmmmm yeah that's a good callout. I can move it